packages/core/src/client.test.ts

```diff
@@ -1265,6 +1265,22 @@ describe('Client', () => {
     expect(client.getCachedReference(createReference(result.entry?.[0]?.resource as Patient))).toBeDefined();
   });
 
+  test('Search and return 404', async () => {
+    console.error = jest.fn();
+    const fetch = mockFetch(404, {
```

Comment:
This isn't testing for the correct thing. In the example the user gave, the response does not have a json representation. Better to do a `mockFetch` that returns a string. 

For example, the following from the original ticket
```
const resp = await fetch('https://api.medplum.com/nonExistingPath/Patient?_count=1')
await resp.text()
```

returns: 
```
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /nonExistingPath/Patient</pre>
</body>
</html>
````